app/schema/api/v4/imports.rb

```diff
@@ -0,0 +1,520 @@
+         resourceType: resource_type("MedicationRequest"),
+         meta: meta,
+         identifier: identifier(description: "ID of medication request"),
+         subject: identifier(description: "Patient ID"),
+         performer: identifier(description: "Facility ID"),
+         medicationReference: {
+           type: :object,
+           properties: {reference: {type: :string, pattern: "^#.*$", nullable: false}},
+           required: ["reference"],
+           description: "This should reference the ID of contained medication resource"
+         },
+         dispenseRequest: {
+           type: [:object, :null],
+           properties: {
+             expectedSupplyDuration: value_quantity(
+               system: "http://unitsofmeasure.org",
+               unit: "days",
+               code: "d"
+             ).merge!(type: [:object, :null], nullable: true)
+           }
+         },
+         dosageInstruction: {
+           type: :object,
+           properties: {
+             timing: {type: :object,
+                      properties: {
+                        code: {type: :string,
+                               enum: %w[QD BID TID QID],
+                               nullable: false,
+                               description: "Mapping to Simple representation\n" \
+                                 "QD: OD (Once a day)\n" \
+                                 "BID: BD (Twice a day)\n" \
+                                 "TID: TDS (Thrice a day)\n" \
+                                 "QID: QDS (Four times a day)"}
+                      }},
+             doseAndRate: {type: "array",
+                           items: {type: :object,
+                                   properties: {
+                                     doseQuantity: value_quantity(
+                                       system: "http://unitsofmeasure.org",
+                                       unit: "mg",
+                                       code: "mg"
+                                     )
+                                   }},
+                           description: "Dosage in milligrams"},
+             text: {type: :string,
+                    description: "Use only if dosage cannot be expressed in terms of the other fields."}
+           }
+         }
+       },
+       required: %w[resourceType meta identifier contained medicationReference subject performer]}
+    end
+
+    def import_request_payload
+      {
+        type: :object,
+        properties: {
+          resources: import_resource_list
+        }
+      }
+    end
+
+    def import_resource_list
+      {
+        type: :array,
+        items: {
+          oneOf: %w[
+            patient
+            appointment
+            observation_blood_pressure
+            observation_blood_sugar
+            condition
+            medication_request
+          ].map { |resource| {"$ref" => "#/definitions/#{resource}"} }
+        }
+      }
+    end
+
+    def all_definitions
```

Comment:
Why not call this simple `definitions`?

Reply:
> Why not call this simply `definitions`?

Mostly for consistency with the [methods returning definitions in other schemas](https://github.com/simpledotorg/simple-server/blob/03c414a330e3520367ba534942eded482a8869b8/app/schema/api/schema.rb#L88-L89). I don't feel strongly about either name for this particular case, so leaving it unchanged for now.

> Also, we can move this def to the top? And make other defs private?

I will move it to the top 👍 

I'd prefer not to make the other methods private because the pattern that I am seeing with these schema definitions is that we are trying to emulate it being Plain Old Data, rather than provide an OOP-class providing some kind of interface + behaviour. If I didn't care about being consistent with the other schemas, I'd have made this a module with module functions rather than a Class with class methods.